doc/stdlib/arrays.rst

```diff
@@ -531,158 +886,260 @@ Combinatorics
 
    Compute the kth lexicographic permutation of a vector.
 
-.. function:: nthperm(p)
+   ::
+
+       nthperm(p)
+
+   Return the "k" that generated permutation "p". Note that "nthperm(nthperm([1:n], k)) == k" for "1 <= k <= factorial(n)".
+
+
+.. function:: nthperm(v, k)
+
+   Compute the kth lexicographic permutation of a vector.
+
+   ::
+
+       nthperm(p)
+
+   Return the "k" that generated permutation "p". Note that "nthperm(nthperm([1:n], k)) == k" for "1 <= k <= factorial(n)".
 
-   Return the ``k`` that generated permutation ``p``.
-   Note that ``nthperm(nthperm([1:n], k)) == k`` for ``1 <= k <= factorial(n)``.
 
 .. function:: nthperm!(v, k)
 
-   In-place version of :func:`nthperm`.
+   In-place version of "nthperm()".
+
 
-.. function:: randperm([rng,] n)
+.. function:: randperm([rng], n)
+
+   Construct a random permutation of length "n". The optional "rng" argument specifies a random number generator, see *Random Numbers*.
 
-   Construct a random permutation of length ``n``. The optional ``rng`` argument
-   specifies a random number generator, see :ref:`Random Numbers <random-numbers>`.
 
 .. function:: invperm(v)
 
    Return the inverse permutation of v.
 
+
 .. function:: isperm(v) -> Bool
 
    Returns true if v is a valid permutation.
 
+
 .. function:: permute!(v, p)
 
-   Permute vector ``v`` in-place, according to permutation ``p``.  No
-   checking is done to verify that ``p`` is a permutation.
+   Permute vector "v" in-place, according to permutation "p".  No checking is done to verify that "p" is a permutation.
+
+   To return a new permutation, use "v[p]".  Note that this is generally faster than "permute!(v,p)" for large vectors.
 
-   To return a new permutation, use ``v[p]``.  Note that this is
-   generally faster than ``permute!(v,p)`` for large vectors.
 
 .. function:: ipermute!(v, p)
 
    Like permute!, but the inverse of the given permutation is applied.
 
-.. function:: randcycle([rng,] n)
 
-   Construct a random cyclic permutation of length ``n``. The optional ``rng``
-   argument specifies a random number generator, see :ref:`Random Numbers
-   <random-numbers>`.
+.. function:: randcycle([rng], n)
+
+   Construct a random cyclic permutation of length "n". The optional "rng" argument specifies a random number generator, see *Random Numbers*.
+
+
+.. function:: shuffle([rng], v)
 
-.. function:: shuffle([rng,] v)
+   Return a randomly permuted copy of "v". The optional "rng" argument specifies a random number generator, see *Random Numbers*.
 
-   Return a randomly permuted copy of ``v``. The optional ``rng`` argument
-   specifies a random number generator, see :ref:`Random Numbers
-   <random-numbers>`.
 
-.. function:: shuffle!([rng,] v)
+.. function:: shuffle!([rng], v)
 
-   In-place version of :func:`shuffle`.
+   In-place version of "shuffle()".
```

Comment:
This won't be sufficient. There are cross-references with renaming in the old ReST documentation with text of the form `f(examplevar)` which are cross-referenced to `f`.